src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -287,7 +287,39 @@ export class TerminalService implements ITerminalService {
 
 	public getAvailableProfiles(): ITerminalProfile[] {
 		this._updateAvailableProfiles();
-		return this._availableProfiles?.map(s => ({ profileName: s.profileName, path: s.path, args: s.args } as ITerminalProfile)) || [];
+		return this._availableProfiles?.map(s => ({ profileName: s.profileName, path: s.path, args: s.args, isWorkspaceProfile: this._getWorkspaceProfilePermissions(s) } as ITerminalProfile)) || [];
+	}
+
+	private _getWorkspaceProfilePermissions(profile: ITerminalProfile): boolean {
+		if (isWindows) {
+			const profiles = this._configurationService.inspect<{ [key: string]: ITerminalProfileObject }>(`terminal.integrated.profiles.windows`);
+			if (profiles && profiles.workspaceValue && profiles.defaultValue) {
+				const workspaceProfile = Object.entries(profiles.workspaceValue).find(p => p[0] === profile.profileName);
+				const defaultProfile = Object.entries(profiles.defaultValue).find(p => p[0] === profile.profileName);
+				if (workspaceProfile && defaultProfile && workspaceProfile[0] === defaultProfile[0]) {
+					let result = !this._terminalProfileObjectEqual(workspaceProfile[1], defaultProfile[1]);
+					return result;
+				} else {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private _terminalProfileObjectEqual(one?: ITerminalProfileObject, two?: ITerminalProfileObject): boolean {
+		if (one === null && two === null) {
+			return true;
+		} else if (one && two && (one as ITerminalExecutable).path && (two as ITerminalExecutable).path) {
+			const oneExec = (one as ITerminalExecutable);
+			const twoExec = (two as ITerminalExecutable);
+			return oneExec.path === twoExec.path && oneExec.args === twoExec.args;
```

Comment:
These are arrays so they need to be checked individually too, you could use `Array.prototype.every` for this to make it compact:

```ts
one.args.every((e, i) => e === two.args[i])
```

The above would only work if you verified the lengths are identical above it.

Reply:
args can be string or string[]

Reply:
will fix tho. just going to be a lot of code

Reply:
path is also of type string | string[]

Reply:
can further narrow for `typeof path === 'string'`